tests/testTriggers.go

```diff
@@ -720,6 +727,207 @@ func TriggerVolumeResize(contexts *[]*scheduler.Context, recordChan *chan *Event
 	})
 }
 
+// TriggerCloudSnapShot deploy cloudsnap apps and validates snapshot
+func TriggerCloudSnapShot(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer ginkgo.GinkgoRecover()
+	uuid := GenerateUUID()
+	event := &EventRecord{
+		Event: Event{
+			ID:   uuid,
+			Type: CloudSnapShot,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+
+	snapshotScheduleRetryInterval := 10 * time.Second
+	snapshotScheduleRetryTimeout := 3 * time.Minute
+
+	Step("Validate Cloud Snaps", func() {
+
+		for _, ctx := range *contexts {
+			var appVolumes []*volume.Volume
+			var err error
+			if strings.Contains(ctx.App.Key, "cloudsnap") {
+
+				Step(fmt.Sprintf("get volumes for %s app", ctx.App.Key), func() {
+					appVolumes, err = Inst().S.GetVolumes(ctx)
+					UpdateOutcome(event, err)
+					if len(appVolumes) == 0 {
+						UpdateOutcome(event, fmt.Errorf("found no volumes for app %s", ctx.App.Key))
+					}
+				})
+				logrus.Infof("Got volume count : %v", len(appVolumes))
+
+				appNamespace := ctx.App.Key + "-" + ctx.UID
+				logrus.Infof("Namespace : %v", appNamespace)
+
+				for _, v := range appVolumes {
+					snapshotScheduleName := v.Name + "-interval-schedule"
+					logrus.Infof("snapshotScheduleName : %v", snapshotScheduleName)
+					snapStatuses, err := storkops.Instance().ValidateSnapshotSchedule(snapshotScheduleName,
+						appNamespace,
+						snapshotScheduleRetryTimeout,
+						snapshotScheduleRetryInterval)
+					if err == nil {
+						for k, v := range snapStatuses {
+							logrus.Infof("Policy Type: %v", k)
+							for _, e := range v {
+								logrus.Infof("ScheduledVolumeSnapShot Name: %v", e.Name)
+								logrus.Infof("ScheduledVolumeSnapShot status: %v", e.Status)
+							}
+						}
+					} else {
+						logrus.Infof("Got error while getting volume snapshot status :%v", err.Error())
+					}
+					UpdateOutcome(event, err)
```

Comment:
This will also cause even `nil` errors to be added to outcome.
Can move `UpdateOutcome(event, err)` to after line 786 inside the `else` part